x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_model_health.tsx

```diff
@@ -0,0 +1,144 @@
+  'xpack.enterpriseSearch.inferencePipelineCard.modelState.stopping',
+  {
+    defaultMessage: 'Stopping',
+  }
+);
+const modelStoppingTooltip = i18n.translate(
+  'xpack.enterpriseSearch.inferencePipelineCard.modelState.stopping.tooltip',
+  {
+    defaultMessage:
+      'This trained model is in the process of shutting down and is currently unavailable',
+  }
+);
+const modelDeploymentFailedText = i18n.translate(
+  'xpack.enterpriseSearch.inferencePipelineCard.modelState.deploymentFailed',
+  {
+    defaultMessage: 'Deployment failed',
+  }
+);
+const modelNotDeployedText = i18n.translate(
+  'xpack.enterpriseSearch.inferencePipelineCard.modelState.notDeployed',
+  {
+    defaultMessage: 'Not deployed',
+  }
+);
+const modelNotDeployedTooltip = i18n.translate(
+  'xpack.enterpriseSearch.inferencePipelineCard.modelState.notDeployed.tooltip',
+  {
+    defaultMessage:
+      'This trained model is not currently deployed. Visit the trained models page to make changes',
+  }
+);
+
+export const TrainedModelHealth: React.FC<InferencePipeline> = ({
+  modelState,
+  modelStateReason,
+}) => {
+  let modelHealth: {
+    healthColor: string;
+    healthText: React.ReactNode;
+    tooltipText: React.ReactNode;
+  };
+  switch (modelState) {
+    case TrainedModelState.Started:
+      modelHealth = {
+        healthColor: 'success',
+        healthText: modelStartedText,
+        tooltipText: modelStartedTooltip,
+      };
+      break;
+    case TrainedModelState.Stopping:
+      modelHealth = {
+        healthColor: 'warning',
+        healthText: modelStoppingText,
+        tooltipText: modelStoppingTooltip,
+      };
+      break;
+    case TrainedModelState.Starting:
+      modelHealth = {
+        healthColor: 'warning',
+        healthText: modelStartingText,
+        tooltipText: modelStartingTooltip,
+      };
+      break;
+    case TrainedModelState.Failed:
+      modelHealth = {
+        healthColor: 'danger',
+        healthText: modelDeploymentFailedText,
+        tooltipText: (
+          <FormattedMessage
+            id="xpack.enterpriseSearch.inferencePipelineCard.modelState.deploymentFailed.tooltip"
+            defaultMessage="The trained model failed to deploy. {reason}"
+            values={{
+              reason: modelStateReason
+                ? i18n.translate(
+                    'xpack.enterpriseSearch.inferencePipelineCard.modelState.deploymentFailed.tooltip.reason',
+                    {
+                      defaultMessage: 'Reason: {modelStateReason}',
+                      values: {
+                        modelStateReason,
+                      },
+                    }
+                  )
+                : '',
+            }}
+          />
+        ),
+      };
+      break;
+    case TrainedModelState.NotDeployed:
+      modelHealth = {
+        healthColor: 'danger',
+        healthText: modelNotDeployedText,
+        tooltipText: modelNotDeployedTooltip,
+      };
+      break;
```

Comment:
Suggestion: Is it worth handling the default case, using the `NotDeployed` modelHealth?